cmd/bedFilter/bedFilter.go

```diff
@@ -7,75 +7,79 @@ import (
 	"fmt"
 	"github.com/vertgenlab/gonomics/bed"
 	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/exception"
 	"github.com/vertgenlab/gonomics/fileio"
 	"github.com/vertgenlab/gonomics/numbers"
 	"log"
-	"strings"
+	"math"
+	"math/rand"
 )
 
-func bedFilter(infile string, outfile string, minScore int, maxScore int, minLength int, maxLength int, minStart int, maxStart int, minEnd int, maxEnd int, chrom string) {
-	var outlist []bed.Bed
-	var line string
-	var startNum, endNum, length int
-	var doneReading bool = false
+func bedFilter(s Settings) {
+	common.RngSeed(s.RandSeed, s.SetSeed)
+	var length int
 	var pass bool = false
-	var numFields int
-	var current bed.Bed
-	file := fileio.EasyOpen(infile)
-	defer file.Close()
+	var r float64
+	bedChan := bed.GoReadToChan(s.InFile)
+	out := fileio.EasyCreate(s.OutFile)
 
-	for line, doneReading = fileio.EasyNextRealLine(file); !doneReading; line, doneReading = fileio.EasyNextRealLine(file) {
+	for curr := range bedChan {
 		pass = true
-		words := strings.Split(line, "\t")
-		startNum = common.StringToInt(words[1])
-		endNum = common.StringToInt(words[2])
-		length = endNum - startNum
-		if len(words) > 4 {
-			if common.StringToInt(words[4]) < minScore {
+		length = curr.ChromEnd - curr.ChromStart
+		if curr.FieldsInitialized > 4 {
```

Comment:
Right now I think records without a score pass a score threshold.  Is this desired?  Maybe there should be an "else" that gives a pass = false?